yappi/_yappi.c

```diff
@@ -1198,7 +1301,7 @@ static void
 _ensure_thread_profiled(PyThreadState *ts)
 {
     if (ts->c_profilefunc != _yapp_callback)
-        _profile_thread(ts);
+        tl_prev_ctx = _profile_thread(ts);

```

Comment:
If there is a way to use a thread-id mapped hash table like ctx table I would prefer using TLS over that. If that is not possible you could do something like following:

```C
#if PY_MINOR_VERSION >= 7
    ptr = PyThread_tss_get(&KEY);
#else
    ptr = PyThread_get_key_value(KEY);
#endif
```

Internal TLS API is changed with Python 3.7.